vmm/src/config.rs

```diff
@@ -2370,8 +2370,19 @@ impl VmConfig {
                 path: PathBuf::from(k),
             });
         }
+
+        let cpus = if let Some(_dtb) = &dtb {
```

Comment:
I think this approach could work...but I think I prefer to omit it for now, and only take things from the device tree that we really need (we can assume that -cpus boot= matches what's in the device tree for now).
i.e. I think any parsing of the device tree, we do in vm::new_craton() for now...

For upstreaming stuff, this might be a good approach, but we need to be careful about what we take from the device tree - e.g. maybe the user wants to specify CPU topology in the CLI, so we need to take the other config fields into account.

Reply:
-cpus boot= matches what's in the device tree for now), >> This is not constant, how do make consistent?

Reply:
Vm struct is initialized based on config. If we don't update the config VM initialises the Vcpus with default that is wrong.

Reply:
The ARE launcher can do it, as it does for qemu:
https://github.com/Azure/azure-sphere-are-launcher/blob/21c0888c07b931e3f7739213c3921194237a8cf3/launcher/src/qemu.rs#L136

Reply:
Right. Thank you for pointing it out.